src/gamepad/gamepad-mappings-manager.vala

```diff
@@ -34,6 +41,12 @@ private class Games.GamepadMappingsManager : Object {
 		add_from_input_stream (resources_open_stream (path, ResourceLookupFlags.NONE));
 	}
 
+	public void add_from_file (string file_name) throws Error {
+		var file = File.new_for_path (file_name);
+		if (!file.query_exists ()) return;
```

Comment:
At this point if the file doesn't exist it should be an error and not silently do nothing. Just remove that line.